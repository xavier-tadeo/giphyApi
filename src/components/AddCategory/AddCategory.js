import React, { useState } from "react";
import "./AddCategory.scss";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("mas");

  const handleInputChange = (evt) => {
    setInputValue(evt.target.value);
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    setCategories((categories) => [...categories, inputValue]);
  };
  return (
    <div className="addcategory">
      <h3 className="addcategory__title">Added category</h3>
      <form className="addcategory__form" onSubmit={handleSubmit}>
        <input
          type="text"
          className="addcategory__input"
          value={inputValue}
          onChange={handleInputChange}
        />
      </form>
    </div>
  );
};
