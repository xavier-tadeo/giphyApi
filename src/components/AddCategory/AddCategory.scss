.addcategory {
  &__title {
    color: blue;
  }

  &__input {
    padding: 20px;
    width: 100%;
  }
}
