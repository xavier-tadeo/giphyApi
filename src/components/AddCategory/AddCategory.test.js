import { AddCategory } from "./AddCategory";
import { shallow } from "enzyme";

describe("Given a AddCategoy component", () => {
  const setCategory = jest.fn();
  let wrapper;
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategory} />);
  });

  describe("When it is render", () => {
    test("It should rendered a h3Tag with name 'Added category' and inputTag", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("When the user writte in the input", () => {
    test("It should change the text in it", () => {
      const value = "Hello word";
      const input = wrapper.find("input");

      input.simulate("change", {
        target: {
          value: value,
        },
      });
    });
  });

  describe("When the user touch enter button without info in input", () => {
    test("It should don t post info", () => {
      wrapper.find("form").simulate("submit", { preventDefault() {} });

      expect(setCategory).not.toHaveBeenCalled();
    });
  });

  describe("When the user touch enter button with info", () => {
    test("It should post info and clear the input", () => {
      const value = "Ball dragon";
      const input = wrapper.find("input");

      input.simulate("change", {
        target: {
          value: value,
        },
      });
      wrapper.find("form").simulate("submit", { preventDefault() {} });

      expect(setCategory).toHaveBeenCalled();
      expect(input.prop("value")).toBe("");
      expect(setCategory).toHaveBeenCalledWith(expect.any(Function));
    });
  });
});
