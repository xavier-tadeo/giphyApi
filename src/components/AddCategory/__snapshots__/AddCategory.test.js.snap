// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a AddCategoy component When it is render It should rendered a h3Tag with name 'Added category' and inputTag 1`] = `
<div
  className="addcategory"
>
  <h3
    className="addcategory__title animate__animated animate__bounce"
  >
    Added category
  </h3>
  <div
    className="addcategory__container-form"
  >
    <img
      alt="buu go out"
      className="addcategory__img"
      src="https://media.giphy.com/media/xTiTnwi8Azjnva46Fq/giphy.gif"
    />
    <form
      className="addcategory__form"
      onSubmit={[Function]}
    >
      <input
        className="addcategory__input"
        onChange={[Function]}
        type="text"
        value=""
      />
    </form>
    <img
      alt="buu talking"
      className="addcategory__img animate__animated animate__hinge"
      src="https://media.giphy.com/media/ZNrjpRCSKbQPe/giphy.gif"
    />
  </div>
</div>
`;
